src/s3ql/backends/backblaze.py

```diff
@@ -0,0 +1,704 @@
+                                      '/b2api/v1/b2_list_file_names',
+                                      self.conn_api,
+                                      body=body)
+
+        j = json.loads(str(body, encoding='UTF-8'))
+        if j['files'][0]['fileName'] == key_with_prefix:
+            return j['files'][0]['contentLength']
+
+        raise NoSuchObject(key_with_prefix)
+
+    @retry
+    @copy_ancestor_docstring
+    def open_read(self, key):
+        log.debug('started with %s', key)
+
+        key = self._encode_key(key)
+
+        resp, data = self._do_request('GET',
+                                      '/file/%s/%s%s' % (self.bucket_name, self.prefix, key),
+                                      self.conn_download, download_body=False)
+
+        meta = self._extractmeta(resp, key)
+
+        return ObjectR(key, self.conn_download, resp, metadata=meta)
+
+    @copy_ancestor_docstring
+    def open_write(self, key, metadata=None, is_compressed=False):
+        log.debug('started with %s', key)
+
+        key = self._encode_key(key)
+
+        return ObjectW(key, self, metadata)
+
+    @retry
+    @copy_ancestor_docstring
+    def clear(self):
+        log.debug('started')
+        for file in self.list():
+            self.delete(file,force=True)
+
+    @retry
+    @copy_ancestor_docstring
+    def delete(self, key, force=False):
+        log.debug('started with %s', key)
+
+        todel_id = self._list_file_version(key)
+
+        if not todel_id and not force:
+            raise NoSuchObject(key)
+
+        self._delete_file_id(todel_id)
+
+    @copy_ancestor_docstring
+    def list(self, prefix=''):
+        log.debug('started with %s', prefix)
+
+        prefix = self._encode_key(prefix)
+
+        next_filename = self.prefix + prefix
+        keys_remaining = True
+
+        while keys_remaining and next_filename is not None:
+            next_filename, filelist = self._list_file_name(next_filename)
+
+            while filelist:
+                file = filelist.pop(0)
+
+                if file.startswith(self.prefix + prefix):
+                    # remove prefix before return
+                    r = file[len(self.prefix):]
+                    yield self._decode_key(r)
+                else:
+                    keys_remaining = False
+                    break
+
+    @prepend_ancestor_docstring
+    def copy(self, src, dest, metadata=None):
+        """No atomic copy operation on backend. Hope this does not append to often"""
+        log.debug('started with %s, %s', src, dest)
+
+        data, src_meta = self.fetch(src)
+
+        # Delete dest file if already exist
+        self.delete(dest, force=True)
+
+        if metadata is None:
+            dst_meta = src_meta
+        else:
+            dst_meta = metadata
+
+        self.store(dest, data, dst_meta)
+
+    @copy_ancestor_docstring
+    def update_meta(self, key, metadata):
+        log.debug('started with %s', key)
+        self.copy(key, key, metadata)
```

Comment:
I think this is expected to be cheap and remote-only. Please review where this is used, you may have to find a different solution.